pkg/controller/watch/watch.go

```diff
@@ -0,0 +1,73 @@
+package watch
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ResourceWatcher implements handler.EventHandler and is used to trigger reconciliation when
+// a watched object changes. It's designed to only be used for a single type of object.
+// If multiple types should be watched, one ResourceWatcher for each type should be used.
+type ResourceWatcher struct {
+	watched map[types.NamespacedName][]types.NamespacedName
+}
+
+// New will create a new ResourceWatcher with no watched objects.
+func New() ResourceWatcher {
+	return ResourceWatcher{
+		watched: make(map[types.NamespacedName][]types.NamespacedName),
+	}
+}
+
+// Add will add a new object to watch.
+func (w ResourceWatcher) Add(watchedName, dependentName types.NamespacedName) {
```

Comment:
[nit] The name `Add` is a bit misleading, I think maybe something like `EnsureAdded` or `EnsureWatched` would be more accurate, what do you think?

Reply:
I agree that `Add` is a bit misleading but I'm not sure including `Ensure` adds much. How about `Watch`? Feels natural that you would `Watch` with a `Watcher`!

Reply:
I think I'm okay with Watch!

Reply:
Renamed!